tck/api/src/main/java/org/eclipse/microprofile/metrics/tck/TagsTest.java

```diff
@@ -22,53 +22,168 @@
     }
 
     @Test
-    public void tagsTest() {
-        Assert.assertNotNull(metadata);
-        Assert.assertTrue(metadata.getTags().containsValue("blue"));
+    @InSequence(1)
+    public void simpleTagTest() {
+        Tag one = new Tag("hello", "world");
+        Tag two = new Tag("goodbye", "friend");
+        MetricID metricID = new MetricID("metricName", one, two);
+        
+        assertThat(metricID.getTags(), hasKey("hello"));
+        assertThat(metricID.getTags(), hasValue("world"));
+        
+        assertThat(metricID.getTags(), hasKey("goodbye"));
+        assertThat(metricID.getTags(), hasValue("friend"));
+        
+        //MP_METRICS_TAGS=tier=integration
+        assertThat("Counter's Global Tag not set properly", metricID.getTags(), hasKey("tier"));
+        assertThat("Counter's Global Tag not set properly", metricID.getTags(), hasValue("integration"));
     }
 
     @Test
-    public void addTagsTest() {
-
-        Metadata metadata2 = Metadata.builder(metadata)
-                .addTags("colour=green,size=medium").addTag("number=5").build();
-
-        Assert.assertNotNull(metadata2);
-        Assert.assertTrue(metadata2.getTags().containsKey("size"));
-        Assert.assertTrue(metadata2.getTags().containsValue("green"));
-        Assert.assertFalse(metadata2.getTags().containsValue("blue"));
-        Assert.assertTrue(metadata2.getTags().containsKey("number"));
+    @InSequence(2)
+    public void lastTagValueTest() {
+        
+        Tag tagColour = new Tag("colour","red");
+        Tag tagColourTwo = new Tag("colour","blue");
+        
+        String counterName = "org.eclipse.microprofile.metrics.tck.TagTest.counter";
+        Counter counter = registry.counter(counterName, tagColour, tagColourTwo);
+        
+        //MetricID that only has colour=blue... the last tag value to be passed in
+        MetricID counterMID = new MetricID(counterName, tagColourTwo);
+        
+        //check the metric is registered
+        assertThat("Counter is not registered correctly", registry.getCounters(), hasKey(counterMID));
+    }
+    
+    @Test
+    @InSequence(3)
+    public void counterTagsTest() {
+        
+        Tag tagEarth = new Tag("planet", "earth");
+        Tag tagRed = new Tag("colour", "red");
+        Tag tagBlue = new Tag("colour", "blue");
+        
+        String counterName = "org.eclipse.microprofile.metrics.tck.TagTest.counterColour";
+        
+        Counter counterColour = registry.counter(counterName);
+        Counter counterRed = registry.counter(counterName,tagEarth,tagRed);
+        Counter counterBlue = registry.counter(counterName,tagEarth,tagBlue);
+        
+        MetricID counterColourMID = new MetricID(counterName);
+        MetricID counterRedMID = new MetricID(counterName, tagEarth,tagRed);
+        MetricID counterBlueMID = new MetricID(counterName, tagEarth,tagRed);
```

Comment:
think this last one was supposed to say...
`MetricID counterBlueMID = new MetricID(counterName, tagEarth,tagBlue);`